File: src/chain-observers/classifiers/feed-price-classifier.ts
import { price } from "@hiveio/wax";
import { CollectorClassifierBase } from "./collector-classifier-base";

export interface IFeedPriceData {
  currentMedianHistory: price;
  marketMedianHistory: price;
  currentMinHistory: price;
  currentMaxHistory: price;
  /**
   * Note: This is a timestamp of the last feed price retrieval, not the actual feed price interval timestamp.
   */
  lastFeedPriceRetrievalTimestamp: Date;
  /**
   * Contains feed price history updated every hour from the latest to the oldest.
   */
  priceHistory: Iterable<price>;
}

export class FeedPriceClassifier extends CollectorClassifierBase<IFeedPriceData> {}

File: python/workerbee/chain_observers/classifiers/feed_price_classifier.py
from __future__ import annotations

from ..classifier_results import FeedPriceClassifierData
from .collector_classifier_base import CollectorClassifierBase


class FeedPriceClassifier(CollectorClassifierBase[FeedPriceClassifierData]):
    """IFeedPriceData:
    current_median_history: price
    market_median_history: price
    current_min_history: price
    current_max_history: price
    last_feed_price_retrieval_timestamp: datetime
    price_history: list[price]
    """

    pass
